src/android/Capture.java

```diff
@@ -377,20 +381,54 @@ else if (resultCode == Activity.RESULT_CANCELED) {
         }
     }
 
-
     public void onAudioActivityResult(Request req, Intent intent) {
-        // Get the uri of the audio clip
-        Uri data = intent.getData();
-        if (data == null) {
-            pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: data is null"));
-            return;
+        Uri srcContentUri = intent.getData();
+
+        // Get file name
+        String srcContentUriString = srcContentUri.toString();
+        String fileName = srcContentUriString.substring(srcContentUriString.lastIndexOf('/') + 1);
+
+        // Create dest file path
+        String tempRoot = cordova.getActivity().getCacheDir().getAbsolutePath();
+        String destFile = tempRoot + "/" + fileName;
+
+        // Create dest file
+        File tmpRootFile = new File(destFile);
+
+        try {
+            // If file exists
```

Comment:
`If file does not exist and successfully created`

Reply:
I can split the try-catch like this.

```java
// Create dest file
File tmpRootFile = new File(destFile);

try {
    if (!tmpRootFile.createNewFile()) {
        // When the file already exists
        pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: failed to create file to application cache directory as it already exists."));
        return;
    }
} catch (IOException e) {
    // When the path is invalid
    pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: failed to create new file to application cache directory due to invalid file path."));
    return;
}

try {
    // Write src file content to dest (new file)
} catch (IOException e) {
    // When it fails to copy content over
    pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: failed to copy recording data to new file in application cache directory."));
    return;
}
```

This will provide better feedback on what fails.

- When the file already exists
- When the path is invalid and could not create a file
- When it fails to copy the src file contents to the dest file (new file).

Alternatively, we use `createTempFile` and do not reuse the filename that was uniquely by the recorder.

```java
// Create dest file
String fileExtension = "." + srcContentUriString.substring(srcContentUriString.lastIndexOf('.') + 1);
File tempRootFile = new File(tempRoot);
File tmpNewFile = null;

try {
    tmpNewFile = File.createTempFile("cdv-media-capture", fileExtension, tempRootFile);
}  catch (IOException e) {
    pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: failed to create new temp file in application cache directory."));
    return;
}

try {
    // Write src file content to dest (new file)
} catch (IOException e) {
    // When it fails to copy content over
    pendingRequests.resolveWithFailure(req, createErrorObject(CAPTURE_NO_MEDIA_FILES, "Error: failed to copy recording data to new file in application cache directory."));
    return;
}
```

Reply:
Why not splitting but balanced, as I see other capture types have runtime exception throwing in some cases, checks on null after it, or so. Again maybe keep some sort of common pattern and not go for a new structure only for Audio?

Reply:
> as I see other capture types have runtime exception throwing in some cases

I think all `RuntimeException` usages have been removed now.

> Again maybe keep some sort of common pattern and not go for a new structure only for Audio?

This PR is only touching the Audio aspect. I foresee further PRs being created to address video and image capture.